app/models/widget.rb

```diff
@@ -6,4 +6,16 @@ class Widget < ApplicationRecord
 
   default_scope { order(position: :asc) }
   scope :enabled, -> { where(enabled: true) }
+  scope :next, ->(current_id) {
+    begin
+      current_position = enabled.index(enabled.find_by(widget_id: current_id))
```

Comment:
Ah ok. What is the benefit of using widget_id over id?

The general rule of thumb is you want a scope to return a scope so they are chainable. I'd expect `Widget.next` to always resolve to a model instance so `def self.next(current_id)` would be preferred as a hint its not chainable.

I this is a good case for adding a spec.

It also looks like this commit may be out of order from the schema change. There are a couple of "fix github sort" one liners floating around. The simplest would be to rebase with each commit marked for reword and making no wording changes.